mergify_engine/gh_pr_fullifier.py

```diff
@@ -103,78 +103,90 @@ def disabled_by_rules(pull, **extra):
                     % filtered[0])
     return None
 
+# NOTE(sileht): Github mergeable_state is undocumented, here my finding by
+# testing and and some info from other project:
+#
+# unknown: not yet computed by Github
+# dirty: pull request conflict with the base branch
+# behind: head branch is behind the base branch (only if strict: True)
+# unstable: branch up2date (if strict: True) and not required status
+#           checks are failure or pending
+# clean: branch up2date (if strict: True) and all status check OK
+#
+# https://platform.github.community/t/documentation-about-mergeable-state/4259
+# https://github.com/octokit/octokit.net/issues/1763
+
+
+# Use enum.Enum and make is serializable ?
+class MergifyState(object):
+    NOT_READY = 0
+    NEED_BRANCH_UPDATE = 10
+    ALMOST_READY = 20
+    READY = 30
+
 
-def compute_weight_and_status(pull, **extra):
-    status_desc = disabled_by_rules(pull, **extra)
+def compute_status(pull, **extra):
+    disabled = disabled_by_rules(pull, **extra)
     reviews_ok, reviews_ko, reviews_required = pull.mergify_engine["approvals"]
-    if status_desc is not None:
-        weight = -1
+    if disabled:
+        mergify_state = MergifyState.NOT_READY
+        github_state = "failure"
+        github_desc = disabled
     elif reviews_ko:
-        weight = -1
-        status_desc = "Change requests need to be dismissed"
+        mergify_state = MergifyState.NOT_READY
+        github_state = "pending"
+        github_desc = "Change requests need to be dismissed"
     elif len(reviews_ok) < reviews_required:
-        weight = -1
-        status_desc = (
+        mergify_state = MergifyState.NOT_READY
+        github_state = "pending"
+        github_desc = (
             "%d/%d approvals required" %
             (len(reviews_ok), reviews_required)
         )
-    elif (pull.mergeable_state == "clean"
-          and pull.mergify_engine["combined_status"] == "success"):
-        # Best PR ever, up2date and CI OK
-        weight = 11
-        status_desc = "Will be merged soon"
     elif pull.mergeable_state in ["clean", "unstable"]:
-        weight = 10
-        status_desc = "Will be merged soon"
+        mergify_state = MergifyState.READY
+        github_state = "success"
+        github_desc = "Will be merged soon"
     elif (pull.mergeable_state == "blocked"
           and pull.mergify_engine["combined_status"] == "pending"):
-        # Maybe clean soon, or maybe this is the previous run
-        # selected PR that we just rebase
-        weight = 10
-        status_desc = "Waiting for CI success"
+        # Maybe clean soon, or maybe this is the previous run selected PR that
+        # we just rebase, or maybe not. But we set the mergify_state to
+        # ALMOST_READY to ensure we do not rebase multiple pull request in //
+        mergify_state = MergifyState.ALMOST_READY
+        github_state = "pending"
+        github_desc = "Waiting for CI success"
```

Comment:
Could we change that to `"Waiting for status checks success"` which is more inline about what we wait (CI is more like a corner case of status check).

Reply:
done